### ISSUE TITLE
Add support for handling EC2 instance rebalance recommendations to amazon-ecs-agent

### ISSUE THREAD

**Issue by schmutze** [aws/ec2-spot-instances-integrations-roadmap#22]
An [EC2 instance rebalance recommendations](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/rebalance-recommendations.html) is a new signal that notifies you when a Spot Instance is at elevated risk of interruption. The signal can arrive sooner than the two-minute Spot Instance interruption notice, giving you the opportunity to proactively manage the Spot Instance.

Customers have told us they'd like the [Amazon ECS Agent](https://github.com/aws/amazon-ecs-agent) to support handling the rebalance recommendation natively to automatically rebalance their containerized workloads running on EC2 Spot Instances.

Please let us know your thoughts!

**Comment by holstvoogd** [comment]
yes please, 2 minutes is way to short to spin up a new instance and drain a bunch of tasks, especially if you have any TCP based services as they ignore all health check settings & take at least 5 minutes to drain and restart
